### ISSUE TITLE
Modifying schema of inherited class

### ISSUE THREAD

**Issue by nomasprime** [fotinakis/swagger-blocks#68]
Not sure how to modify inherited schema so I've created them separately. Please could someone show me how to refactor:

``` ruby
swagger_schema :'Users::Registrations::New' do
  property :data do
    key :type, :object

    property :type do
      key :type, :string
      key :default, "users-registrations"
    end

    property :attributes do
      key :required, [:email]
      key :type, :object

      property :email do
        key :type, :string
      end
    end
  end
end

swagger_schema :'Users::Registrations::Existing' do
  property :data do
    key :type, :object
    key :required, [:id]

    property :type do
      key :type, :string
      key :default, "users-registrations"
    end

    property :id do
      key :type, :integer
    end

    property :attributes do
      key :required, [:email]
      key :type, :object

      property :email do
        key :type, :string
      end
    end
  end
end

```

**Comment by JohnSmall** [comment]
Yes, I know the problem. I want to be able to define things in a parent class, and add things in  descendant class. But I can't figure out how to do it. So like you I end up with very non-DRY code, which is bad news.

One thing I'm looking at is being able to pass the containing class into Swagger::Blocks::Node so that I have access to the class that's being documented inside the swagger_schema, property & etc. That way I might be able to reflect on the containing class and get associations, attributes and so on that I want to document. If I can do that then I'll be able to have the swagger_schema in a root class, and it'll get the class name and associations on the descendant classes without having to write repetitive code. 

I'll try this out soon and post a PR.

**Comment by jodigiordano** [comment]
You could do something like that:

    module UserRegistrations
      def self.compile(opts = {})
        Class.new do
          include Swagger::Blocks

          swagger_schema opts[:name] do
            property :data do
              key :type, :object

              if opts[:id]
                key :required, [:id]

                property :id do
                  key :type, :integer
                end
              end

              property :type do
                key :type, :string
                key :default, "users-registrations"
              end

              property :id do
                key :type, :integer
              end

              property :attributes do
                key :required, [:email]
                key :type, :object

                property :email do
                  key :type, :string
                end
              end
            end
          end
        end
      end
    end

Usage:

    UserRegistrations.compile(name: 'Users::Registrations::New', id: false)
    UserRegistrations.compile(name: 'Users::Registrations::Existing', id: true)

No need for inheritance or modifying the lib.
